website/data-storage/src/programs/programs.service.ts

```diff
@@ -90,28 +89,21 @@ export class ProgramsService {
       });
       return program;
     } catch (error) {
-      logger.error(error, error.stack, '');
-      return null;
+      throw new ProgramNotFound();
     }
   }
 
   async setStatus(id: string, genesis: string, status: InitStatus): Promise<IProgram> {
     return new Promise((resolve) => {
       setTimeout(async () => {
-        const program = await this.findProgram({ id, genesis });
-        if (program) {
+        try {
+          const program = await this.findProgram({ id, genesis });
           program.initStatus = status;
           resolve(await this.programRepo.save(program));
+        } catch (error) {
+          errorLog.error(error, 101);
```

Comment:
looks like magic number

Reply:
it's a string number) but maybe you are right